lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+
+  if array1.length == array2.length
```

Comment:
Nit: nested `if`s are harder to understand at a glance. How about something like:
```
if array1.length != array2.length
  return false
end 

# array lengths presumed equal, because of above "guarding" if statement

...
```